src/routes/scope-selection/ScopeSelectionScreen.tsx

```diff
@@ -1,44 +1,68 @@
+import HorizontalLine from '../../components/HorizontalLine'
 import RouteWrapper from '../../components/RouteWrapper'
+import SearchBar from '../../components/SearchBar'
 import ServerResponseHandler from '../../components/ServerResponseHandler'
-import { ContentSecondary } from '../../components/text/Content'
+import { ContentSecondary, ContentTextBold, ContentTextLight } from '../../components/text/Content'
 import { Heading } from '../../components/text/Heading'
 import { BUTTONS_THEME } from '../../constants/data'
 import { Discipline } from '../../constants/endpoints'
+import { formatDiscipline } from '../../hooks/helpers'
+import { useLoadAllDisciplines } from '../../hooks/useLoadAllDisciplines'
 import { useLoadDisciplines } from '../../hooks/useLoadDisciplines'
 import useReadSelectedProfessions from '../../hooks/useReadSelectedProfessions'
 import { RoutesParams } from '../../navigation/NavigationTypes'
-import { setSelectedProfessions } from '../../services/AsyncStorage'
-import { getLabels } from '../../services/helpers'
+import { pushSelectedProfession, setSelectedProfessions } from '../../services/AsyncStorage'
+import { getLabels, splitTextBySearchString } from '../../services/helpers'
 
-const DisciplineContainer = styled.View`
-  margin: 0 ${props => props.theme.spacings.sm};
+const HighlightContainer = styled.Text<{ disabled: boolean }>`
+  flex-direction: row;
+  color: ${props => (props.disabled ? props.theme.colors.disabled : props.theme.colors.black)};
 `
 
-const ButtonContainer = styled.View`
-  margin: ${props => props.theme.spacings.md} auto;
+const TextContainer = styled.View`
+  margin-top: ${props => props.theme.spacings.xxl};
+  margin-bottom: ${props => props.theme.spacings.lg};
 `
 
 const StyledText = styled(ContentSecondary)`
   text-align: center;
 `
 
-const TextContainer = styled.View`
-  margin-top: ${props => props.theme.spacings.xxl};
-  margin-bottom: ${props => props.theme.spacings.lg};
+const SearchContainer = styled.View`
+  margin: ${props => props.theme.spacings.sm};
 `
 
-const StyledScrollView = styled.ScrollView`
-  background-color: ${props => props.theme.colors.background};
+const ScopeContainer = styled.View`
+  margin: 0 ${props => props.theme.spacings.sm};
 `
 
+const ButtonContainer = styled.View`
+  margin: ${props => props.theme.spacings.md} auto;
+`
+
+export const searchProfessions = (disciplines: Discipline[] | undefined, searchKey: string): Discipline[] | undefined =>
+  disciplines?.filter(discipline =>
+    normalize(discipline.title).toLowerCase().includes(normalize(searchKey.toLowerCase()))
+  )
+
+const highlightText = (textArray: [string] | [string, string, string], disabled: boolean): JSX.Element => (
+  <HighlightContainer disabled={disabled}>
+    <ContentTextLight>{textArray[0]}</ContentTextLight>
+    <ContentTextBold>{textArray[1]}</ContentTextBold>
+    <ContentTextLight>{textArray[2]}</ContentTextLight>
+  </HighlightContainer>
+)
```

Comment:
Can we extract this along with the call of `splitTextBySearchString(profession.title, searchString)` to a separate `TextHighlighter` component or similar?

Reply:
I'm not sure that's necessary since the only place (at least that I can think of) that might re-use this is the word search, and that doesn't have any highlighting. So I'd consider an extraction to a separate component a premature optimization. What do you think?

Reply:
Hmm for me its mostly about readability/modularity. But its a well meant suggestion so I am also okay with not doing that here. Just thought it reduces the complexity of this file and adheres to classic design principles.